internal/driver/browser_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package driver
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/chromedp/chromedp"
+)
+
+func maybeSkipBrowserTest(t *testing.T) {
+	// Limit to just Linux for now since this is expensive and the
+	// browser interactions should be platform agnostic.  If we ever
+	// see a benefit from wider testing, we can relax this.
+	if runtime.GOOS != "linux" || runtime.GOARCH != "amd64" {
+		t.Skip("This test only works on x86-64 Linux")
+	}
+
+	// Check that browser is available.
+	if _, err := exec.LookPath("google-chrome"); err == nil {
+		return
+	}
+	if _, err := exec.LookPath("chrome"); err == nil {
+		return
+	}
+	t.Skip("chrome not available")
```

Comment:
FWIW I paused here for a second wondering if this should fail instead of skip so that we know early if the CI gets misconfigured somehow and starts silently skipping the test. But you added a chrome check in the CI configuration so probably we are good.

Reply:
I think we indeed will need to skip here instead of failing the test, because otherwise the test will always fail in environments that do not have chrome, e.g. windows, mac (where we currently skip this test).

Reply:
There is an earlier check that Skips unless we are on Linux+amd64. So we could easily change this one to Fail() which will reduce the long distance coupling to the workflow definition.